conformance/lang/expressions/trap_expr.balt

```diff
@@ -0,0 +1,351 @@
+    io:println(e1); // @output error("foo panics")
+}
+
+Test-Case: output
+Description: Test the result of the trap expression being `v`,
+             when evaluation of the expression does not complete abruptly with panic.
+Labels: error-type, function-call-expr, function-defn, int, nil-literal, nil-type, optional-type, trap-expr, union-type, variable-reference-expr
+
+function init() {
+    int? nilVal = ();
+    int|error? e = trap nilVal;
+    io:println(e is ()); // @output true
+}
+
+Test-Case: error
+Description: If type of expr is T, then type of trap expr is T|error.
+Labels: error-type, int, local-init-var-decl-stmt, multiplicative-expr, nil-literal, optional-type, string, trap-expr, union-type, variable-reference-expr
+
+function errorFunction() {
+    int zero = 0;
+    int? nilVal = ();
+    int _ = trap 2 / zero; // @error incompatible types: expected 'int', found '(int|error)'
+    string|error? x = trap nilVal; // @error incompatible types: expected '(string|error)?', found 'int?'
+    int _ = trap 2; // @error incompatible types: expected 'int', found '(int|error)'
+    string _ = trap "str"; // @error incompatible types: expected 'string', found '(string|error)'
+    float _ = trap 2.3f; // @error incompatible types: expected 'float', found '(float|error)'
+}
+
+Test-Case: error
+Description: If type of expr is T, then type of trap expr is T|error.
+Fail-Issue: ballerina-platform/ballerina-lang#36472
+Labels: error-type, int, local-init-var-decl-stmt, multiplicative-expr, trap-expr, variable-reference-expr
+
+function errorFunction() {
+    int zero = 0;
+    error e = trap 2 / zero; // @error incompatible types: expected 'error', found 'int'
+}
+
+Test-Case: output
+Description: Test trap expression when the type of expression is never.
+Labels: boolean, error-constructor-expr, error-type, function-call-expr, function-defn, int, never, panic, return-stmt, trap-expr, union-type
+
+function funcNeverReturns() returns never {
+    panic error("msg");
+}
+
+function init() {
+    never|error e1 = trap funcNeverReturns();
+    io:println(e1); // @output error("msg")
+    error e2 = trap funcNeverReturns();
+    io:println(e2); // @output error("msg")
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a nil-literal and it does not panic.
+Labels: error-type, is-expr, nil-literal, nil-type, null, optional-type, trap-expr, union-type
+
+function init() {
+    ()|error t1 = trap ();
+    io:println(t1 is ()); // @output true
+
+    error? t2 = trap null;
+    io:println(t2 is ()); // @output true
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a boolean-literal and it does not panic.
+Labels: boolean, boolean-literal, error-type, trap-expr, union-type
+
+function init() {
+    boolean|error t = trap false;
+    io:println(t); // @output false
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a numeric-literal and it does not panic.
```

Comment:
When does a numeric literal panic?

Reply:
Same for string.

Reply:
Thoughts on this? 

Let's also remove the productions from the descriptions everywhere.

numeric-literal -> numeric literal

Reply:
Fixed in 5303f62061d4c9f05d3292185a6988ae3261a1a4